server.go

```diff
@@ -240,6 +239,16 @@ func (s *Server) CreateHandler() http.Handler {
 	return s.rootHandler(s.AroundRequest(s.requestHandler()))
 }
 
+func (s *Server) createResponseHandler() http.Handler {
+	handler := withCombinedFragments(s)
+	handler = withDefaultErrorHandler(handler)
+	handler = s.AroundResponse(handler)
+	handler = multiplexer.WithCombinedServerTimingHeader(handler)
+	handler = multiplexer.WithDefaultHeaders(handler)
```

Comment:
@BlakeWilliams this is the area of most interest now in this PR, this is the middleware stack we are using for generating the response.

The configure `AroundResponse` handler sits between initially setting the default response headers from the first fragment response and the handlers to write the default error response or combined fragment response.

So not calling `next` in the `AroundResponse` handler would prevent viewproxy from writing any response body and allow consumers to implement it (replacing the need for `OnError`).

Reply:
Nice, I think this makes a lot of sense and cleans things up pretty nicely (like removing `onError`).